processor.go

```diff
@@ -119,6 +122,7 @@ func newProcessor(params processorParams) *processor {
 		shutdownTimeout:   params.shutdownTimeout,
 		starting:          params.starting,
 		finished:          params.finished,
+		shuffleRand:       rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
`rand.NewSource` is not thread-safe ([code](https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/math/rand/rng.go;l=238-252)).

This is fine now since only `queues()` uses it, and `queues()` is only called in a single single-threaded context.

But this seems bug-prone, as it would be tempting to re-use it in a different context.

Maybe naming the field `queueShuffleRand` and adding a comment about thread-safety is enough.

Reply:
I thought the comment I added to the field should be clarifying enough – I renamed it since it makes it clearer.

Thanks!